test/factories/host_related.rb

```diff
@@ -167,9 +180,20 @@ def set_nic_attributes(host, attributes, evaluator)
     end
 
     trait :on_compute_resource do
+      transient do
+        expect_queue_compute true
+      end
       uuid Foreman.uuid
       association :compute_resource, :factory => :ec2_cr
-      after(:build) { |host| host.expects(:queue_compute) }
+      after(:build) { |host, evaluator| host.expects(:queue_compute) if evaluator.expect_queue_compute }
```

Comment:
Or perhaps change this to `before(:create)` so it's only expected if the host is going to be saved, and won't be expected when it's built only.